pkg/cli/admin/catalog/build.go

```diff
@@ -0,0 +1,216 @@
+	o := NewBuildImageOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "build",
+		Short: "Builds a registry container image from a collection operator manifests.",
+		Long: templates.LongDesc(`
+			Builds a registry container image from a collection operator manifests.
+
+			Extracts the contents of a collection of operator manifests to disk, and builds them into
+			an operator registry image.
+		`),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.StringVar(&o.From, "from", o.From, "The image to use as a base.")
+	flags.StringVar(&o.To, "to", o.To, "The Docker repository tag to upload the completed catalog image to.")
+	flags.StringVar(&o.AuthToken, "auth-token", "", "Auth token for communicating to appregistry.")
+	flags.StringVar(&o.AppRegistryEndpoint, "app-registry", o.AppRegistryEndpoint, "Endpoint for pulling from an appregistry instance.")
+	flags.StringVarP(&o.AppRegistryNamespace, "namespace", "n", "", "Namespace to pull from an appregistry instance")
+
+	return cmd
+}
+
+type BuildImageOptions struct {
+	genericclioptions.IOStreams
+
+	From, To             string
+	AuthToken            string
+	AppRegistryEndpoint  string
+	AppRegistryNamespace string
+}
+
+func (o *BuildImageOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+
+	if o.To == "" {
+		return fmt.Errorf("you must specify a tag for the resulting image. example: --to=quay.io/myorg/myimage:1.0.3")
+	}
+	if o.From == "" {
+		return fmt.Errorf("you must specify a base image. example: --from=quay.io/operator-framework/upstream-registry-builder:v1.1.0")
+	}
+	if o.AppRegistryEndpoint == "" {
+		return fmt.Errorf("app-registry must be a valid app-registry endpoint")
+	}
+	if o.AppRegistryNamespace == "" {
+		return fmt.Errorf("namespace must be specified")
+	}
+
+	return nil
+}
+
+func (o *BuildImageOptions) Run() error {
+	opts := apprclient.Options{Source: o.AppRegistryEndpoint}
+	if o.AuthToken != "" {
+		opts.AuthToken = o.AuthToken
+	}
+	client, err := apprclient.New(opts)
+	if err != nil {
+		return fmt.Errorf("couldn't connect to appregistry, %s", err.Error())
+	}
+
+	downloader := NewDownloader(client)
+	dir, err := downloader.DownloadManifestsTmp(o.AppRegistryNamespace)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("downloaded to %s\n", dir)
+
+	archivePath, err := BuildDatabaseLayer(dir)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("archive: %s\n", archivePath)
+
+	a := imgappend.NewAppendImageOptions(o.IOStreams)
+	a.From = o.From
+	a.To = o.To
+	a.LayerFiles = append(a.LayerFiles, archivePath)
+	return a.Run()
+}
+
+func BuildDatabaseLayer(manifestPath string) (string, error) {
+	dbDir, err := BuildDatabase(manifestPath)
+	if err != nil {
+		return "", err
+	}
+	return BuildLayer(dbDir, "")
+}
+
+func BuildDatabase(manifestPath string) (string, error) {
+	dbDir, err := ioutil.TempDir("", "db-")
```

Comment:
What's happening with that database file, I don't see it being cleaned. Is it the result of this invocation then I guess a user should be given an option where that file is being written, if it's temporary it should be cleaned.